### PR TITLE
Check for animation styles on close.

### PR DESCRIPTION
**Pull request by gonsfx**
Hello!

We had some problems with vex modals not closing on closeById.
Turns out the problem occured because vex assumes that `$vex` element is styled with an animation.

If it isn't, but the browser supports animations, the animationend event is waited for indefinitely and the modal never actually closed.

I added a check for animation styles on the `$vex` element to fix this and also bumped the Vex version to `2.2.2`.

### CODE REVIEW

**Comment by gonsfx** [comment]
This is in open status for quite a while now. What are your thoughts?

File: coffee/vex.coffee [hunk 1]
```diff
@@ -150,7 +150,19 @@ vexFactory = ($) ->
                 $('body').trigger 'vexAfterClose', options # Triggered on the body since $vexContent was removed
                 options.afterClose $vexContent, options if options.afterClose
 
-            if animationEndSupport
+            hasAnimation = ->
+                $clone = $ $vex[0].cloneNode false
```
Comment (adamschwartz) [inline coffee/vex.coffee]: This is the only step that mildly concerns me. My initial reaction is do we really need to clone the node and all of its contents? Perhaps all we need to do is create a test element with the same inline attributes (classes, id, data attributes, inline styles, etc.) and make it a sibling.

**Comment by adamschwartz** [comment]
@gonsfx Thanks so much for this PR. If we can address my comment above I think this will be ready to merge.

File: coffee/vex.coffee (hunk 1 above)
Comment (gonsfx) [inline coffee/vex.coffee]: I don't care as long as it works.
Could you check your assumption? ;)

**Comment by mkorablin** [comment]
Since `vex` relies on jquery, we can do

    $vex.css('animationName') != 'none' && $vex.css('animationDuration') != '0s'

This gives the benefit of checking vendor specific styles as well. For example, `.css('animationName') will check `webkitAnimationName` if you are in a webkit-based browser.

Alternatively, if jquery for some reason is not an option, we can fall back on pure browser:

    var style = window.getComputedStyle($vex.get(0)),
        hasAnimation = style.animationName != 'none' && style.animationDuration != '0s' && style.mozAnimationName != 'none' && style.animationDuration != '0s' ...

It also would be nice to check for transitions and not just animations.

I don't write coffeScript, so this is all in JS. If you want, I can write up a PR to fix this issue.

**Comment by adamschwartz** [comment]
@mkorablin Thanks for the suggestion and offer. Please feel free. :+1:

**Comment by mkorablin** [comment]
@adamschwartz Do you write in CoffeeScript and translate it to JS? Or should I update both?

**Comment by adamschwartz** [comment]
We write it in CoffeeScript and then run Grunt in order to build the JS and other compiled files. If you’d like to do that just run `npm install`, `bower install`, and `npm install -g grunt-cli`, and then `grunt` to actually build the project. If you’d prefer to do the PR in JS though that’s totally fine. We can convert it to Coffee as a final step. So whatever you prefer. :smile:

**Comment by mkorablin** [comment]
@adamschwartz, sorry for being silent for so long. Finally created a pull request to resolve this: https://github.com/HubSpot/vex/pull/139

**Comment by bbatliner** [comment]
Seems like #139 solved this issue. Going ahead and cleaning up this open PR, in that case. Reopen if needed...

**Pull request closed without merging**